core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -480,6 +484,15 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
   }
 }
 
+/**
+ * Helper class for the cleanable segment of a log and whether to update the checkpoint associated with the log
+ * firstDirtyOffset the lower (inclusive) and firstUncleanableDirtyOffset the upper(exclusive) offsets
```

Comment:
Could you use `@param` tag for each parameter description? When describing `needUpdateCheckpoint`, would be useful to mention that `firstDirtyOffset` is the new checkpoint offset to update to (if true).

Also, there may be multiple segments between `firstDirtyOffset` and `firstUncleanableDirtyOffset` offsets, so instead of "for the cleanable segment of a log" maybe say something like "that represents a range of dirty offsets that can be cleaned".